prow/tide/tide.go

```diff
@@ -51,6 +54,36 @@ type Controller struct {
 	ghc    githubClient
 	kc     kubeClient
 	gc     *git.Client
+
+	m     sync.Mutex
+	pools []Pool
+}
+
+type Action string
+
+const (
+	Wait         Action = "WAIT"
+	Trigger             = "TRIGGER"
+	TriggerBatch        = "TRIGGER_BATCH"
+	Merge               = "MERGE"
+	MergeBatch          = "MERGE_BATCH"
+)
+
+type Pool struct {
+	Org    string
+	Repo   string
+	Branch string
```

Comment:
Do you need JSON annotations for this struct?

Reply:
I don't *think* this will be externally exposed, but if it is then we'll have to add them.